From: Chad Monroe <chad@monroe.io>
Date: Mon, 24 Nov 2025 14:34:21 -0800
Subject: [PATCH] wifi-scripts: add acs_num_scans hostapd config option

Support adjusting the number of scans made during ACS.

Declare acs_num_scans in the wifi-device schema and emit it from the
device config generator in usr/share/ucode/wifi/hostapd.uc.

Fixes: OWRT-8792

Signed-off-by: Chad Monroe <chad@monroe.io>
---
--- a/src/usr/share/schema/wireless.wifi-device.json
+++ b/src/usr/share/schema/wireless.wifi-device.json
@@ -13,6 +13,10 @@
 			"type": "boolean",
 			"default": false
 		},
+		"acs_num_scans": {
+			"description": "Number of scans to perform during ACS",
+			"type": "number"
+		},
 		"airtime_mode": {
 			"description": "Set the airtime policy operating mode",
 			"type": "number",
--- a/src/usr/share/ucode/wifi/hostapd.uc
+++ b/src/usr/share/ucode/wifi/hostapd.uc
@@ -529,7 +529,7 @@ function generate(config) {
 	else
 		set_default(config, 'enable_background_radar', false);
 
-	append_vars(config, [ 'acs_chan_bias', 'acs_exclude_dfs', 'enable_background_radar' ]);
+	append_vars(config, [ 'acs_chan_bias', 'acs_exclude_dfs', 'acs_num_scans', 'enable_background_radar' ]);
 
 	/* TX Power */
 	append_vars(config, [ 'min_tx_power' ]);
